RecoBTag/PixelCluster/plugins/PixelClusterTagInfoProducer.cc

```diff
@@ -0,0 +1,357 @@
+//        double detZ = theGeomDet->surface().position().z();
+        int layer = 0; // 1-4
+        
+        if(subid==1) {  // barrel
+            PixelBarrelName pbn(detid, tTopo, m_isPhase1);
+            layer = pbn.layerName();
+        }
+        else if(m_addFPIX && subid==2) {  // forward
+            PixelEndcapName pen(detid, tTopo, m_isPhase1);
+            layer = pen.diskName();
+        }
+        if(layer == 0 || layer > m_nLayers) continue;
+//        std::cout << layer << "\t" << detX << " : " << detUnit->x() << "\t" << detY << " : " << detUnit->y() << "\t" << detZ << " : " << detUnit->z() << std::endl;
+        
+        for(edmNew::DetSet<SiPixelCluster>::const_iterator clustIt = detUnit->begin(); clustIt != detUnit->end(); ++clustIt) {
+            // get global position of the cluster
+            LocalPoint lp = topol->localPosition(MeasurementPoint(clustIt->x(), clustIt->y()));
+            GlobalPoint clustgp = theGeomDet->surface().toGlobal( lp );
+            if(m_minADC > 0  and clustIt->charge() < m_minADC) continue;
+            reco::PixelClusterProperties cp = { clustgp.x(), clustgp.y(), clustgp.z(), clustIt->charge(), layer };
+//            clusterX.push_back(clustgp.x());
+//            clusterY.push_back(clustgp.y());
+//            clusterZ.push_back(clustgp.z());
+//            clusterC.push_back(clustIt->charge());
+            clusters.push_back(cp);
+        }
+//    if( numberOfClusters < 1.) continue;
+//    
+//    detUnit_layer .push_back( layer ); 
+//    detUnit_X         .push_back(detX     );
+//    detUnit_Y         .push_back(detY     );
+//    detUnit_Z         .push_back(detZ     );
+//    
+//    nClusters.push_back(numberOfClusters);
+//    cluster_globalz  .push_back(_cluster_globalz);
+//    cluster_globalx  .push_back(_cluster_globalx);
+//    cluster_globaly  .push_back(_cluster_globaly);
+//    cluster_charge   .push_back(_cluster_charge);
+    
+    }
+//    std::cout << clusterC.size() << std::endl;
+   
+    // Loop over jets
+    for(unsigned int j = 0, nj = collectionJets->size(); j < nj; j++) {
+        if(collectionJets->at(j).pt() < m_minJetPt) continue;
+
+        edm::RefToBase<reco::Jet> jetRef = collectionJets->refAt(j);
+//        reco::PixelClusterData* data = new reco::PixelClusterData();
+//        reco::PixelClusterTagInfo* tagInfo = new reco::PixelClusterTagInfo();
+
+        reco::PixelClusterData data = {}; // Initialize new Data to 0
+        reco::PixelClusterTagInfo tagInfo = {};
+
+        
+        for(auto cluIt = clusters.begin(); cluIt != clusters.end(); ++cluIt) {
+            TVector3 c3(cluIt->x - firstPV->x(), cluIt->y - firstPV->y(), cluIt->z - firstPV->z());
+            TVector3 j3(jetRef->px(), jetRef->py(), jetRef->pz());
+            float dR = j3.DeltaR(c3);
+            float sC = 12. * 2. / (jetRef->pt()); // * jetRef->correctedP4(0).pt()
+//            fillData(data, dR, sC, cluIt->layer, cluIt->charge);
```

Comment:
Prima di fare la PR i commenti vanno rimossi

Reply:
Si, certo. Ci sono anche dei cout che mi servono per capire se il codice gira. Tutto da togliere per la PR, ma si fa in pochi minuti.